### PR TITLE
chore: Remove OSS from fdroid build

### LINKED ISSUE #1255: Remove Play Services dependencies from fdroid flavor
### Chore

**Description**
To launch the app in fdroid the play services dependencies will be removed. Once the app is released on fdroid, we can focus upon their replacement.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #1255 
Checklist:

- [ ] I have checked for PMD and check-style issues
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: 
The OSS library dependency was remove from Fdroid release.

### CODE REVIEW

**Comment by sridharjajoo** [comment]
@iamareebjamal Is this fine for OSS libraries? And for other libraries should we follow a similar approach i.e hiding ?

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 1]
```diff
@@ -65,8 +66,9 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
             return true;
         });
 
+        findPreference(getString(R.string.acknowledgements_key)).setVisible(new AcknowledgementDecider().shouldShowAcknowledgement());
         findPreference(getString(R.string.acknowledgements_key)).setOnPreferenceClickListener(preference -> {
-            startActivity(new Intent(getActivity(), OssLicensesMenuActivity.class));
+            new AcknowledgementDecider().openAcknowledgementsSection(getActivity());
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: Why instantiating 2 times? Create a final member

**Comment by iamareebjamal** [comment]
You could inject context in constructor and inject the component in fragment then

**Comment by sridharjajoo** [comment]
@iamareebjamal Locally,I injected the AcknowledgementDecider constructor with Context parameter
```
public class AcknowledgementDecider {
    private final Context context;
    @Inject
    public AcknowledgementDecider(Context context) {
        this.context = context;
    }
```
 and passed getActivity( ) in it from the Settings Fragment but the app is crashing on selecting Acknowledgement button. Is that what was to be done ?

File: app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java [hunk 2]
```diff
@@ -65,8 +67,9 @@ public void onCreatePreferencesFix(@Nullable Bundle bundle, String rootKey) {
             return true;
         });
 
+        findPreference(getString(R.string.acknowledgements_key)).setVisible(acknowledgementDecider.shouldShowAcknowledgement());
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/settings/SettingsFragment.java]: OK, fix indentation and we'll go ahead with this

**Comment by iamareebjamal** [comment]
Also, I think there'll still be problem because of the applied plugin of oss library

**Comment by sridharjajoo** [comment]
I tried testing it in fdroiddebug mode. It was working fine.
Will the problem be because fdroid might not accept it?

**Comment by iamareebjamal** [comment]
Yes

Also, the igenius maven repository is unacceptable

**Comment by iamareebjamal** [comment]
Right now go ahead with this

Strangely, on my local fdroid, application built without error even with the GMS libraries, don't know how, so just replace this, stetho-noop, places and vision library and try if that works for F-Droid guys

**Comment by iamareebjamal** [comment]
There's a bug in F-Droid due to which it doesn't detect the libraries, I'm sending the Merge Request

**Pull request merged**